docs/kap_proposals/kap_6_hashicorp_vault.md

```diff
@@ -0,0 +1,136 @@
+# Hashicorp Vault  
+  
+This feature allows the user to fetch secrets from [Hashicorp Vault](https://www.vaultproject.io/), an online Kapitan Secret backend `vault`.  
+  
+Author: [@vaibahvk](https://github.com/vaibhavk) [@daminisatya](https://github.com/daminisatya)  
+## Specification  
+  
+The following variables need to be exported to the environment(depending on authentication used) where you will run `kapitan secrets --reveal` in order to authenticate to your HashiCorp Vault instance:  
+* VAULT_ADDR: URL for vault  
+* VAULT_SKIP_VERIFY=true: if set, do not verify presented TLS certificate before communicating with Vault server. Setting this variable is not recommended except during testing  
+* VAULT_TOKEN: token for vault or file (~/.vault-tokens)  
+* VAULT_ROLE_ID: (required by approle)  
+* VAULT_SECRET_ID: (required by approle)  
+* VAULT_USERNAME: username to login to vault  
+* VAULT_PASSWORD: password to login to vault  
+* VAULT_CLIENT_KEY: the path to an unencrypted PEM-encoded private key matching the client certificate  
+* VAULT_CLIENT_CERT: the path to a PEM-encoded client certificate for TLS authentication to the Vault server  
+* VAULT_CACERT: the path to a PEM-encoded CA cert file to use to verify the Vault server TLS certificate  
+* VAULT_CAPATH: the path to a directory of PEM-encoded CA cert files to verify the Vault server TLS certificate  
+* VAULT_NAMESPACE: specify the Vault Namespace, if you have one  
+  
+Considering a key-value pair like `my_key`:`my_secret` ( in our case let’s store `hello`:`batman` inside the vault ) in the path `secret/foo` on the vault server, to use this as a secret either follow:  
+  
+```shell  
+$ echo “{'path':'secret/foo','key':'hello'}” > somefile.txt  
+$ kapitan secrets --write vault:path/to/secret_inside_kapitan --file somefile.txt --target dev-sea --auth userpass 
+```  
+or in a single line  
+```shell  
+$ echo “{'path':'secret/foo','key':'hello'}” | kapitan secrets --write vault:path/to/secret_inside_kapitan -t dev-sea -a userpass -f -  
+```  
+The entire string __{'path':'secret/foo','key':'hello'}__ is base64 encoded and stored in the secret_inside_kapitan. Now secret_inside_kapitan contains the following  
+  
+```yaml    
+data: 4oCccGF0aDpzZWNlcnQvZm9v4oCdIOKAnGtleTpteV9rZXnigJ0K  
+encoding: original  
+parameter:
```

Comment:
Added commit for loading parameters from inventory.
Any suggestions on how to test it? Since testing on temporary directory wouldn't work and now revealing the secret requires `--target` so testing the CLI require an example `vaultkv` secret in either of the `examples/` that we already have.